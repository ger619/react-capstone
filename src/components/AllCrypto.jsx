import React, { useState, useEffect } from 'react';
import millify from 'millify';
import { Link } from 'react-router-dom';
import {
  Card, Row, Col, Input,
} from 'antd';
import { useGetCryptosQuery } from '../services/cryptoApi';
// eslint-disable-next-line react/prop-types
const AllCrypto = ({ simplified }) => {
  const count = simplified ? 10 : 100;
  const { data: cryptosList, isFetching } = useGetCryptosQuery(count);
  const [cryptos, setCryptos] = useState(cryptosList?.data?.coins);
  const [searchTerm, setSearchTerm] = useState('');
  useEffect(() => {
    // eslint-disable-next-line max-len
    const filterData = cryptosList?.data?.coins?.filter((coin) => coin.name.toLowerCase().includes(searchTerm.toLowerCase()));
    setCryptos(filterData);
  }, [cryptosList, searchTerm]);

  if (isFetching) return <div>Loading...</div>;
  return (
    <>
      {!simplified && (
        <div className="search-crypto">
          <Input placeholder="Search Cryptocurrencies" onChange={(e) => setSearchTerm(e.target.value)} />
        </div>
      )}

      <Row gutter={[28, 28]} className="crypto-card-container">
        {cryptos?.map((crypto) => (
          <Col xs={24} sm={12} lg={6} className="crypto-card" key={crypto.id}>
            <Link to={`/crypto/${crypto.id}`}>
              <Card
                title={`${crypto.rank}.${crypto.name}`}
                extra={<img className="crypto-image" src={crypto.iconUrl} style={{ width: 48 }} alt="Crypto" />}
                hoverable
              >
                <p>
                  Price:
                  {millify(crypto.price)}
                </p>
                <p>
                  Market Cap:
                  {millify(crypto.marketCap)}
                </p>
                <p>
                  Daily Change:
                  {millify(crypto.change)}
                  %
                </p>
              </Card>
            </Link>
          </Col>
        ))}
      </Row>
    </>
  );
};

export default AllCrypto;
