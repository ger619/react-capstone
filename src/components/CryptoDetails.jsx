import React from 'react';
import HTMLReactParser from 'html-react-parser';
import { useParams } from 'react-router-dom';
import millify from 'millify';
import {
  Col, Row, Typography,
} from 'antd';
import {
  // eslint-disable-next-line max-len
  MoneyCollectOutlined, DollarCircleOutlined, FundOutlined, ExclamationCircleOutlined, StopOutlined, TrophyOutlined, CheckOutlined, NumberOutlined, ThunderboltOutlined,
} from '@ant-design/icons';
import { useGetCryptoDetailsQuery } from '../services/cryptoApi';
import './styles/CryptoDetails.css';

const { Title, Text } = Typography;

const CryptoDetails = () => {
  const { uuid } = useParams();
  const { data, isFetching } = useGetCryptoDetailsQuery(uuid);
  const cryptoDetails = data?.data?.coin;

  if (isFetching) return <div>Loading...</div>;

  const stats = [
    { title: 'Price to USD', value: `$ ${cryptoDetails?.price && millify(cryptoDetails?.price)}`, icon: <DollarCircleOutlined /> },
    { title: 'Rank', value: cryptoDetails?.rank, icon: <NumberOutlined /> },
    { title: '24h Volume', value: `$ ${cryptoDetails?.volume && millify(cryptoDetails?.volume)}`, icon: <ThunderboltOutlined /> },
    { title: 'Market Cap', value: `$ ${cryptoDetails?.marketCap && millify(cryptoDetails?.marketCap)}`, icon: <DollarCircleOutlined /> },
    { title: 'All-time-high(daily avg.)', value: `$ ${cryptoDetails?.allTimeHigh?.price && millify(cryptoDetails?.allTimeHigh?.price)}`, icon: <TrophyOutlined /> },
  ];

  const genericStats = [
    { title: 'Number Of Markets', value: cryptoDetails?.numberOfMarkets, icon: <FundOutlined /> },
    { title: 'Number Of Exchanges', value: cryptoDetails?.numberOfExchanges, icon: <MoneyCollectOutlined /> },
    { title: 'Approved Supply', value: cryptoDetails?.supply?.confirmed ? <CheckOutlined /> : <StopOutlined />, icon: <ExclamationCircleOutlined /> },
    { title: 'Total Supply', value: `$ ${cryptoDetails?.supply?.total && millify(cryptoDetails?.supply?.total)}`, icon: <ExclamationCircleOutlined /> },
    { title: 'Circulating Supply', value: `$ ${cryptoDetails?.supply?.circulating && millify(cryptoDetails?.supply?.circulating)}`, icon: <ExclamationCircleOutlined /> },
  ];

  return (
    <Col className="coin-details-container">
      <Row>
        <Col className="coin-heading-container">
          <Title level={2} className="coin-name">
            {cryptoDetails?.name}
            {' '}
            Price
          </Title>
          <p>
            Live price in USD View value statistics for
            {' '}
            {cryptoDetails?.name}
            {' '}
            market cap, volume, and more.
          </p>

        </Col>
      </Row>
      <Row>
        <Col className="stats-container">
          <Col className="coin-value-heading">
            <Title level={3} className="coin-value-stats-title">
              {cryptoDetails?.name}

              {' '}
              Value Statistics
            </Title>
            <p>
              View value statistics for
              {cryptoDetails?.name}
              {' '}
              market cap, volume, and more.
            </p>
          </Col>
          {stats.map(({ icon, title, value }) => (
            // eslint-disable-next-line react/jsx-key
            <Col className="coin-stats">
              <Col className="stat-stat-name">
                <Text>{icon}</Text>
                <Text>{title}</Text>
                <Text className="stats">{value}</Text>
              </Col>
            </Col>
          ))}
        </Col>
        <Col className="stats-container">
          <Col className="coin-value-heading">
            <Title level={3} className="coin-value-stats-title">
              Other Statistics
            </Title>
            <p>
              View value statistics for All Cryptocurrencies.
              market cap, volume, and more.
            </p>
          </Col>
          {genericStats.map(({ icon, title, value }) => (
            // eslint-disable-next-line react/jsx-key
            <Col className="coin-stats">
              <Col className="stat-stat-name">
                <Text>{icon}</Text>
                <Text>{title}</Text>
                <Text className="stats">{value}</Text>
              </Col>
            </Col>
          ))}
        </Col>
      </Row>
      <Col className="coin-desk-link">
        <Row>
          <Title level={3} className="coin-details-heading">
            What is
            {' '}
            {cryptoDetails?.name}
          </Title>
          <Text>{HTMLReactParser(cryptoDetails?.description) }</Text>
        </Row>

      </Col>

    </Col>
  );
};
export default CryptoDetails;
