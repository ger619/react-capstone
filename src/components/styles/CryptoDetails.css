.coin-details-container {
  padding: 1rem;
}
