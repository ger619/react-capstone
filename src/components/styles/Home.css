.home {
  padding: 1rem;
}

.heading {
  text-align: center;
}

.home-heading-container {
  display: flex;
  font-size: 1.5rem;
  margin-top: 1rem;
}

.home-heading-container p {
  padding-right: 2rem;
}

.home-heading-container p:nth-child(even) {
  float: right;
  font-size: 1rem;
  text-align: right;
  padding-top: 1rem;
}
