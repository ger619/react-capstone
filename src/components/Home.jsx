import React from 'react';
import {
  Typography, Row, Col, Statistic,
} from 'antd';
import millify from 'millify';
import { Link } from 'react-router-dom';
import { useGetCryptosQuery } from '../services/cryptoApi';
import AllCrypto from './AllCrypto';

const { Title } = Typography;

const Home = () => {
  const { data, isFetching } = useGetCryptosQuery(10);
  const globalStats = data?.data?.stats;
  if (isFetching) return <div>Loading...</div>;
  return (
    <>
      <Title level={2} className="heading"> Ger Crypto Stats</Title>
      <Row>
        <Col span={12}><Statistic title="All Crypto Currencies" value={globalStats.total} /></Col>
        <Col span={12}><Statistic title="Total Exchanges" value={globalStats.totalExchanges} /></Col>
        <Col span={12}><Statistic title="Total Market Cap" value={millify(globalStats.totalMarketCap)} /></Col>
        <Col span={12}><Statistic title="Total Volume" value={millify(globalStats.total24hVolume)} /></Col>
        <Col span={12}><Statistic title="Total Markets" value={millify(globalStats.totalMarkets)} /></Col>
      </Row>
      <div className="home-heading-container">
        <Title level={2} className="home-title"> Top 10 Cryptocurrencies</Title>
        <Title level={3} className="show-more"><Link to="/allCrypto">Show More Coins</Link></Title>
      </div>
      <AllCrypto simplified />
    </>
  );
};
export default Home;
