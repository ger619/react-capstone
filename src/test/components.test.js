import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../components/Home';
import AllCrypto from '../components/AllCrypto';
import CryptoDetails from '../components/CryptoDetails';

describe('Test Components', () => {
  test('test Home', () => {
    const tree = renderer
      .create(<Provider store={store}><Home /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test AllCrypto', () => {
    const tree = renderer
      .create(<Provider store={store}><AllCrypto /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test CryptoDetails', () => {
    const tree = renderer
      .create(<Provider store={store}><CryptoDetails /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
