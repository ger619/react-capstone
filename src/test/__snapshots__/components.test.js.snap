// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Components test AllCrypto 1`] = `
<div>
  Loading...
</div>
`;

exports[`Test Components test CryptoDetails 1`] = `
<div>
  Loading...
</div>
`;

exports[`Test Components test Home 1`] = `
<div>
  Loading...
</div>
`;
